sdk/python/kfp/components/structures/components.json_schema.json

```diff
@@ -0,0 +1,402 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
```

Comment:
Please reduce the schema to the minimum needed for pipeline component sharing.

Reply:
Please, note that there is always a balance between the feature set and size. E.g. full Kubernetes properties support is not needed for our 1st party sample components or the components that the contractors will build. However removing that support would be controversial.

For the purpose of incremental code review I can remove `GraphSpec` and `GraphImplementation` classes from the spec for now, but removing `TaskSpec` is problematic since it contains the full k8s Container spec and Pod spec.

Reply:
Also note that in other review comments the features you ask will multiply the size of the spec.
K8s spec is ~100.000 lines.

Reply:
Can't you refer to the k8s spec with a $ref pointing to another URI instead?

Reply:
> Can't you refer to the k8s spec with a $ref pointing to another URI instead?

I do exactly that when it's possible. See https://github.com/kubeflow/pipelines/blob/618b559391a225a6d821141047d6f7438e6f64a1/sdk/python/kfp/components/structures/components.json_schema.json#L364

However any change or addition requires copying parts of the K8s spec.

BTW, Even Kubernetes team decided not to support full K8s spec when they createdtheir `PodTemplate`. They understand that most settings do not make sense in a template.

Reply:
I have switched to just referring to Kubernetes spec everywhere. There is no mentions of Argo anymore.